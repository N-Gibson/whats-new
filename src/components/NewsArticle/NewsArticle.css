* {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
}

.article {
  background-color: #fff;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  margin: 10px;
  text-align: center;
  width: 46%;
}

a {
  background-color: #B56AF0;
  color: #fff;
  height: 45px;
  margin: 10px 0px 0px 0px;
  padding: 15px 0px 10px 30px;
  text-align: left;
  text-decoration: none;
  width: 100%;
}