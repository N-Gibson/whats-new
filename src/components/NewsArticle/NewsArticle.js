import React from 'react';
import './NewsArticle.css';

const NewsArticle = (story) => {
  return (
  <article className='article'>
    <img src={story.story.img}/>
    <h2>{story.story.headline}</h2>
    <p>{story.story.description}</p>
    <a href={story.story.url}>Link To Article</a>
  </article>
  )
}

export default NewsArticle;