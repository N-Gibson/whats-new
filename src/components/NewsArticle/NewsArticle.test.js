import React from 'react';
import { shallow } from 'enzyme';
import NewsArticle from './NewsArticle';

describe('News Article', () => {
  it('should match the snapshot', () => {
    const mockArticle = {
      id: 1,
      headline: 'mock headline',
      description: 'mock description',
      img: 'mock image filepath',
      url: 'mock url'
    }
    const wrapper = shallow(<NewsArticle story={mockArticle}/>)

    expect(wrapper).toMatchSnapshot();
  });
});