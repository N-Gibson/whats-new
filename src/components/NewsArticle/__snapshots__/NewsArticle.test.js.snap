// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`News Article should match the snapshot 1`] = `
<article
  className="article"
>
  <img
    src="mock image filepath"
  />
  <h2>
    mock headline
  </h2>
  <p>
    mock description
  </p>
  <a
    href="mock url"
  >
    Link To Article
  </a>
</article>
`;
